core/src/main/java/com/scalar/db/api/DistributedStorageMetadata.java

```diff
@@ -0,0 +1,208 @@
+package com.scalar.db.api;
+
+import com.google.common.base.MoreObjects;
+import java.util.Objects;
+import javax.annotation.concurrent.Immutable;
+
+/** Metadata for a distributed storage. */
+@Immutable
+public class DistributedStorageMetadata {
+
+  /** The type of the storage. */
+  private final String type;
+
+  /**
+   * The name of the storage. If you use `multi-storage`, it is the storage name configured in your
+   * multi-storage configuration. Otherwise, it is the same as the type.
+   */
+  private final String name;
+
+  /** Whether the linearizable scan all is supported or not. */
+  private final boolean linearizableScanAllSupported;
```

Comment:
Could `linearizableScanAllSupported` for storage be false even if the storage can do it but doesn't implement it?
I first thought `linearizableScanAllSupported` is the current implementation status, so it could be false for storage that can do a linearizable scan but doesn't implement the ScanAll yet.
If it is the property of storage, then I'm wondering if this should be unified with the atomicity level property since linearizability is basically atomicity.
Let's discuss it as well 🙇

Reply:
Ah, in that sense, `linearizableScanAllSupported` could be implementation status. I thought it was the property of the storage implementation. The term `status` sounds to me that it could be changed in the future.

Maybe my focus wasn't so much on the underlying storages/databases (e.g., DynamoDB, MySQL) themselves, but rather on the storage implementations (the `DistributedStorage` implementations). Thus, I considered `atomicityLevel` and `linearizableScanAllSupported` as properties of the storage implementations.

> If it is the property of storage, then I'm wondering if this should be unified with the atomicity level property since linearizability is basically atomicity.

Although atomicity and linearizability are technically different concepts, for the implementation perspective, they are basically similar. Therefore, that's a valid idea. Let's discuss it later. Thanks.